### PR TITLE
Simple detection of common Azure and Azure AD secrets

### PR DESCRIPTION
**Pull request by bittrance**
It is tricky to correctly detect sensitive information in the Azure context since 1) sensitive information has few distinguishing marks and 2) there are variations across the MS ecosystem. This PR catches two common classes:
- "credential chain" environment variables as per https://learn.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential, https://github.com/Azure/azure-sdk-for-rust/blob/882e895509b07b633dc80b086c4c6bc5b9dd8867/sdk/identity/src/token_credentials/environment_credentials.rs#L7 and a bazillion other variations
- Likely JSON keys and variable names such as adTenantId, azure_client_id, client_secret so long as they are followed by likely sensitive information.

There are prolly many other likely variation. For example, it seems some corners of the MS ecosystem refers to Client ID as "App ID", but I fear this is a bit too generic to target.

Fixes: #386 .

### CODE REVIEW

**Review by slarwise — COMMENTED** [review]
Awesome, looks good to me :+1:

File: packages/@secretlint/secretlint-rule-azure/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ]
+}
+```
+
+## MessageIDs
+
+### AzureTenantId
+
+> found Azure AD tenant ID: {{ID}}
+
```
Comment (azu) [inline packages/@secretlint/secretlint-rule-azure/README.md:30]: Can you add short description to each message Ids?

Or, just add a link to [EnvironmentCredential Class (Azure.Identity) - Azure for .NET Developers | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential).

Secretlint Lint error show the link to here.

File: packages/@secretlint/secretlint-rule-azure/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+### AzureClientId
+
+> found Azure client id: {{ID}}
+
+### AzureClientSecret
+
+> found Azurre client secret: {{SECRET}}
```
[the remark this answers is not in the document]
Comment (bittrance) [reply packages/@secretlint/secretlint-rule-azure/README.md:48]: Fixed.

File: packages/@secretlint/secretlint-rule-azure/README.md (hunk 1 above)
Comment (bittrance) [reply packages/@secretlint/secretlint-rule-azure/README.md:36]: Something like this, perhaps? Also added some general links at the top.

Comment (azu) [reply packages/@secretlint/secretlint-rule-azure/README.md:36]: Nice!

**Comment by azu** [comment]
Thanks!
https://github.com/secretlint/secretlint/releases/tag/v6.2.0 released 🎉

**Pull request merged**